pkg/kube/inject/inject.go

```diff
@@ -572,6 +572,29 @@ func InjectionData(sidecarTemplate, valuesConfig, version string, typeMetadata *
 		"toLower":             strings.ToLower,
 	}
 
+	// Migrating from values.yaml to mesh or env (for istiod - set using fromEnv instead of values)
+	// This allows an optional config map with vendor-specific settings/overrides - alternative is creating
+	// vendor-specific injection templates or a huge mess.
+	funcMap["value"] = func(key string, def string) string {
+		fromEnv := os.Getenv(strings.ReplaceAll(key, ".", "_"))
+		if fromEnv != "" {
+			return fromEnv
+		}
+		// Special case for 'Values.global' - where most settings are
+		if !strings.Contains(key, ".") {
+			global := values["global"]
+			if globalM, ok := global.(map[string]interface{}); ok {
```

Comment:
this won't work if you have a nested value like `foo.bar`, right?

Reply:
Well, it will work if I find the code used by galley/cobra/etc - having env variables override cli and yaml is already supported somewhere.  

Again - it's not intended to be the final solution, istio/api changes is the right path and happy to remove this code as soon as we get the needed changes.

Reply:
After rename - this is gone. 

Also: the istio/api change adds 'ProxyConfig.env' - after it is merged I'll start using it directly, with same semantics.